providers/component-protocol/cpregister/register_comp_provider.go

```diff
@@ -53,7 +55,12 @@ func RegisterProviderComponent(scenario, componentName string, providerPtr Compo
 	providerName := cputil.MakeComponentProviderName(scenario, componentName)
 
 	// register component
-	RegisterComponent(scenario, componentName, func() cptype.IComponent { return providerPtr })
+	RegisterComponent(scenario, componentName, func() cptype.IComponent {
```

Comment:
separate a func like `wrapXxx`, and add some ut

Reply:
adjusted